tesseract_kinematics/include/tesseract_kinematics/kdl/kdl_utils.h

```diff
@@ -185,68 +182,88 @@ inline bool parseSceneGraph(KDLChainData& results,
     }
     results.robot_chain.addChain(sub_chain);
   }
-  results.tip_name = chains.back().second;
+  results.tip_link_name = chains.back().second;
 
-  results.joint_list.resize(results.robot_chain.getNrOfJoints());
-  results.limits.joint_limits.resize(results.robot_chain.getNrOfJoints(), 2);
-  results.limits.velocity_limits.resize(results.robot_chain.getNrOfJoints());
-  results.limits.acceleration_limits.resize(results.robot_chain.getNrOfJoints());
+  results.data.joint_names.resize(results.robot_chain.getNrOfJoints());
+  results.data.limits.joint_limits.resize(results.robot_chain.getNrOfJoints(), 2);
+  results.data.limits.velocity_limits.resize(results.robot_chain.getNrOfJoints());
+  results.data.limits.acceleration_limits.resize(results.robot_chain.getNrOfJoints());
 
-  results.segment_index[results.base_name] = 0;
-  results.link_list.push_back(results.base_name);
-  results.active_link_list.clear();
-  bool found = false;
+  results.segment_index[results.base_link_name] = 0;
+  results.data.link_names.clear();
+  results.data.active_link_names.clear();
+  results.data.link_names.push_back(results.base_link_name);
+  results.data.active_link_names.push_back(results.base_link_name);
+
+  std::vector<std::string> full_active_link_names;
+  bool found{ false };
   for (unsigned i = 0, j = 0; i < results.robot_chain.getNrOfSegments(); ++i)
   {
     const KDL::Segment& seg = results.robot_chain.getSegment(i);
     const KDL::Joint& jnt = seg.getJoint();
-    results.link_list.push_back(seg.getName());
+    results.data.link_names.push_back(seg.getName());
 
     // KDL segments does not contain the the base link in this list. When calling function that take segmentNr, like
     // JntToCart to get the base link transform you would pass an index of zero and for subsequent links it is
     // index + 1. This was determined through testing which is captured in this packages unit tests.
     results.segment_index[seg.getName()] = static_cast<int>(i + 1);
 
     if (found)
-      results.active_link_list.push_back(seg.getName());
+      results.data.active_link_names.push_back(seg.getName());
 
     if (jnt.getType() == KDL::Joint::None)
       continue;
 
     if (!found)
     {
       found = true;
-      results.active_link_list.push_back(seg.getName());
+      results.data.active_link_names.push_back(seg.getName());
     }
```

Comment:
is `active_link_names` set here? 
If you want to make active_link_names in the right order, should I do changes to this file?

Reply:
I don't think the active links should need to be in the same order. If the check is requiring this then it should be updated.